src/commands/setup.cr

```diff
@@ -1,72 +1,95 @@
+    end
 
-        # Create mtenv-controlled shims for the Myst binary.
-        puts "Linking shims to #{shims_location}"
-        myst_shim_path = File.expand_path("~/.mtenv/shims/myst")
-        myst_install_path = File.join(shims_location, "myst")
-        case
-        when File.symlink?(myst_install_path)
-          existing_path = File.real_path(myst_install_path)
-          if existing_path != myst_shim_path
-            STDERR.puts "`myst` already exists in #{shims_location} and points to #{existing_path}"
-            STDERR.puts "Remove it and re-run this setup to allow `mtenv` to install a new shim."
-            exit(1)
-          end
-        when File.exists?(myst_install_path)
-          STDERR.puts "`myst` already exists in #{shims_location} as a plain file."
-          STDERR.puts "Remove it and re-run this setup to allow `mtenv` to install a new shim."
-          exit(1)
-        else
-          File.symlink(myst_shim_path, myst_install_path)
-        end
-      end
+    private def init_mtenv_home(shims_location)
+      puts "Initializing mtenv home: `#{MTENV.home}`..."
+      FileUtils.mkdir_p "versions"
+      FileUtils.mkdir_p "shims"
+      FileUtils.touch   "global"
+      # Store the shims bindir location for use in the future (e.g., `implode`).
+      File.write filename: "shims_dir", content: shims_location
 
-      puts "\nmtenv setup finished successfully."
+      puts "Creating shims..."
+      File.open("shims/myst", mode: "w", perm: 0o755) do |f|
+        f.truncate
+        f.puts(MYST_SHIM)
+      end
     end
 
+    private def link_shims(shims_location)
+      puts "Linking shims to #{shims_location}"
+
+      myst_shim_path    = MTENV.from_home "shims/myst"
+      myst_install_path = File.join(shims_location, "myst")
+
+      case
+      when File.symlink?(myst_install_path)
+        existing_path = File.real_path(myst_install_path)
+        if existing_path != myst_shim_path
+          abort <<-MSG
+            `myst` already exists in #{shims_location} and points to #{existing_path}
+            Remove it and re-run this setup to allow `mtenv` to install a new shim.
+          MSG
+        end
+      when File.exists?(myst_install_path)
+        abort <<-MSG
+          `myst` already exists in #{shims_location} as a plain file.
+          Remove it and re-run this setup to allow `mtenv` to install a new shim.
+        MSG
+      else
+        File.symlink(myst_shim_path, myst_install_path)
+      end
+    end
 
     private def ask_for_shims_location!
-      print "Where should mtenv create links to shims? (default '/usr/local/bin'): "
-      location = gets
+      # First check if mtenv was passed a --shims-dir flag
+      if (path = flags.shims_dir)
+        path = File.expand_path path
+        (Dir.exists? path)    || Util.fail! "Passed shims-dir does not exist!"
+     ###(File.writable? path) || Util.fail! "mtenv lacks write access to shims-dir" # TODO: Find out whether or not to include this check
```

Comment:
This seems like a good check to have. Not sure if the `TODO` has to do with compatibility or anything else, but I would encourage having this since mtenv wouldn't be able to function on a directory it can't write to.

Reply:
Absolutely!

Reply:
And there seems to be a second and related `TODO` just below, I'll take care of that one as well.